Allow FilingSGML to be created from text of submission in addition to source url/path

cthamilton opened the pull request:
If I read the source code correctly, there is no way to generate a FilingSGML from a string without saving it to a file first. I thought I'd try to make a PR with a small change to from_source method for creating a FilingSGML instance that allows for a string with the full document text to be used also.

dgunning reviewed (commented):
Thanks for this, much needed. To disambiguate the parameters they need to be keyword only in case content is passed without keywords

```python. e.g 
from_source(cls, *, source: Union[str, Path, None]=None, content: Union[str, None]=None) -> 'FilingSGML':
```

cthamilton commented:
Thanks! Good point. I updated the PR as suggested.

dgunning commented:
I agree. It's probably better to add a separate function. It seems that `SGML.from_source` is being actively used and making  it keyword only might break things.

dgunning commented:
Hey Clint, any update on this? I can take over and credit you with the change if that's ok?

cthamilton commented:
Hi Dwight, I implemented the solution I was thinking of. Basically just make the parsing of from _source it's own function which generates the header and documents and also added a new classmethod which basically just applies the function to text.

If you think a different solution would be better feel free to take over. I was making this PR because it would be very directly helpful in my research. (And probably would be to others using the WRDS SEC repository which has the full text submissions cached essentially.)

dgunning reviewed (approved):
Thanks for this PR, and your patience in making these changes.

Also thanks for the link to the Wharton SEC filings archive.

I have done some other changes to this file so will merge.

cthamilton commented:
Thanks! Unfortunately, I realized that the WRDS filings are slightly cleaned and this appears to create some unexpected issues. I may try to introduce a PR to handle these if I'm able to figure out the issues and you think it is within scope. The issues in the header aren't immediately obvious so I'll have to investigate further. The header was giving me an issue somehow related to `COMPANY DATA`.

As an example WRDS style header:
`	<SEC-DOCUMENT>0001213900-24-107406.txt : 20241210
	<SEC-HEADER>0001213900-24-107406.hdr.sgml : 20241210
	<ACCEPTANCE-DATETIME>20241210160508
	ACCESSION NUMBER:		0001213900-24-107406
	CONFORMED SUBMISSION TYPE:	8-K
	PUBLIC DOCUMENT COUNT:		15
	CONFORMED PERIOD OF REPORT:	20241205
	ITEM INFORMATION:		Entry into a Material Definitive Agreement
	ITEM INFORMATION:		Unregistered Sales of Equity Securities
	ITEM INFORMATION:		Other Events
	ITEM INFORMATION:		Financial Statements and Exhibits
	FILED AS OF DATE:		20241210
	DATE AS OF CHANGE:		20241210

	FILER:

		COMPANY DATA:	
			COMPANY CONFORMED NAME:			Marpai, Inc.
			CENTRAL INDEX KEY:			0001844392
			STANDARD INDUSTRIAL CLASSIFICATION:	SERVICES-MISC HEALTH & ALLIED SERVICES, NEC [8090]
			ORGANIZATION NAME:           	08 Industrial Applications and Services
			IRS NUMBER:				000000000
			STATE OF INCORPORATION:			DE
			FISCAL YEAR END:			1231

		FILING VALUES:
			FORM TYPE:		8-K
			SEC ACT:		1934 Act
			SEC FILE NUMBER:	001-40904
			FILM NUMBER:		241538468

		BUSINESS ADDRESS:	
			STREET 1:		615 CHANNELSIDE DRIVE
			STREET 2:		SUITE 207
			CITY:			TAMPA
			STATE:			FL
			ZIP:			33602
			BUSINESS PHONE:		646-303-3483

		MAIL ADDRESS:	
			STREET 1:		615 CHANNELSIDE DRIVE
			STREET 2:		SUITE 207
			CITY:			TAMPA
			STATE:			FL
			ZIP:			33602
	</SEC-HEADER>`

versus raw header:
`<SEC-DOCUMENT>0001213900-24-107406.txt : 20241210
<SEC-HEADER>0001213900-24-107406.hdr.sgml : 20241210
<ACCEPTANCE-DATETIME>20241210160508
ACCESSION NUMBER:		0001213900-24-107406
CONFORMED SUBMISSION TYPE:	8-K
PUBLIC DOCUMENT COUNT:		15
CONFORMED PERIOD OF REPORT:	20241205
ITEM INFORMATION:		Entry into a Material Definitive Agreement
ITEM INFORMATION:		Unregistered Sales of Equity Securities
ITEM INFORMATION:		Other Events
ITEM INFORMATION:		Financial Statements and Exhibits
FILED AS OF DATE:		20241210
DATE AS OF CHANGE:		20241210

FILER:

	COMPANY DATA:	
		COMPANY CONFORMED NAME:			Marpai, Inc.
		CENTRAL INDEX KEY:			0001844392
		STANDARD INDUSTRIAL CLASSIFICATION:	SERVICES-MISC HEALTH & ALLIED SERVICES, NEC [8090]
		ORGANIZATION NAME:           	08 Industrial Applications and Services
		IRS NUMBER:				000000000
		STATE OF INCORPORATION:			DE
		FISCAL YEAR END:			1231

	FILING VALUES:
		FORM TYPE:		8-K
		SEC ACT:		1934 Act
		SEC FILE NUMBER:	001-40904
		FILM NUMBER:		241538468

	BUSINESS ADDRESS:	
		STREET 1:		615 CHANNELSIDE DRIVE
		STREET 2:		SUITE 207
		CITY:			TAMPA
		STATE:			FL
		ZIP:			33602
		BUSINESS PHONE:		646-303-3483

	MAIL ADDRESS:	
		STREET 1:		615 CHANNELSIDE DRIVE
		STREET 2:		SUITE 207
		CITY:			TAMPA
		STATE:			FL
		ZIP:			33602
</SEC-HEADER>`

cthamilton commented:
Attempted to update with your changes to from_source since I started the PR.
